src/main/java/org/jetbrains/kotlinx/lincheck/strategy/managed/TracePoint.kt

```diff
@@ -260,6 +260,7 @@ internal enum class SwitchReason(private val reason: String) {
  */
 internal class CallStackTraceElement(val call: MethodCallTracePoint, val identifier: Int)
 
+@Synchronized
```

Comment:
`objectNumeration` can be cleared concurrently in the `appendTrace` in `TraceReporter.kt`.
It is a bug and its simplest solution is to add `synchronized(objectNumeration)` for the whole `appendTrace` method.
We do not need additional synchronized here, because this method is called only from `appendTrace` (perhaps, we need a note about it?)